packages/cli/src/lib/decorators/preconditions/node.ts

```diff
@@ -0,0 +1,64 @@
+import {constants} from 'os';
+import * as dedent from 'dedent';
+
+import type Command from '@oclif/command';
+import {valid as isVersionValid, lt as isVersionLessThan} from 'semver';
+
+import {SpawnProcessOutput, spawnProcessOutput} from '../../utils/process';
+
+const linkToReadme =
+  'https://github.com/coveo/cli/wiki/Node.js,-NPM-and-NPX-requirements';
+
+export function IsNodeVersionAbove(requiredVersion: string) {
+  return async function (target: Command) {
+    if (!isVersionValid(requiredVersion)) {
+      target.warn(dedent`
+        Required version invalid: "${requiredVersion}".
+        Please report this error to Coveo: https://github.com/coveo/cli/issues/new
+      `);
```

Comment:
If this warning is triggered, there's a big chance that it's not the user machine that ain't properly set up, but that we messed up and didn't call properly the function.

Hopefully, this code will never run, but if it does, at least the user will know what to do (contact us so we can fix it).

Reply:
It wouldn't hurt to have this check. 👍